lib/src/frontend/future_matchers.dart

```diff
@@ -80,3 +80,57 @@ class _Completes extends AsyncMatcher {
     return description;
   }
 }
+
+/// Matches a [Future] that does not complete.
+///
+/// Note that this creates an asynchronous expectation. The call to
+/// `expect()` that includes this will return immediately and execution will
+/// continue. Later, after 20  passes through the event queue,
+/// the matcher will complete.
+///
+/// This returns an [AsyncMatcher], so [expect] won't complete until
+/// [timesToPump] passes through the event queue.
+final Matcher doesNotComplete = const _DoesNotComplete(20);
+
+/// Matches a [Future] that does not complete.
+///
+/// Note that this creates an asynchronous expectation. The call to
+/// `expect()` that includes this will return immediately and execution will
+/// continue. Later, after [timesToPump]  passes through the event queue,
+/// the matcher will complete.
+///
+/// This returns an [AsyncMatcher], so [expect] won't complete until
+/// [timesToPump] passes through the event queue.
+Matcher doesNotCompleteAfter(int timesToPump) =>
+    new _DoesNotComplete(timesToPump);
+
+class _DoesNotComplete extends AsyncMatcher {
+  final int timesToPump;
+  const _DoesNotComplete(this.timesToPump);
+
+  Future _pumpEventQueue(times) {
+    if (times == 0) return new Future.value();
+    return new Future(() => _pumpEventQueue(times - 1));
+  }
+
+  /*FutureOr<String>*/ matchAsync(item) {
+    if (item is! Future) return null;
+
+    var value;
+
+    return new Future(() async {
```

Comment:
`new Future(() async{` feels weird to me. What is our goal there? Do we need it to wait both an event loop (new Future) and a microtask (async)?

If we don't need to double up we can either:

```dart
return () async {
    ...
   await _pumpEventQueue(timeToPump);
  return notCompleted ? null : 'completed with a value of $value';
}();
```

or

```dart
var value;
var notCompleted = true;
item.then((value) {
  value = value;
  notCompleted = false;
});
return _pumpEventQueue(timeToPump)
    .then((_) => notCompleted ? null : 'completed with a value of $value');
```